posts/Yoann/2023-07-07-hibernate-search-6-2-0-Final.adoc

```diff
@@ -0,0 +1,522 @@
+) {
+}
+----
+<1> Projects to the entity identifier (explicit projection through an explicit annotation)
+<2> Projects to field `title` (implicit projection)
+
+Explicit annotations also allow configuring field projections more precisely,
+for example setting the path of a field projection to something else
+than just the name of the constructor parameter:
+
+[source, JAVA, indent=0]
+----
+@ProjectionConstructor
+public record MyBookTitleAndAuthorNamesProjection(
+        @FieldProjection // <1>
+        String title,
+        @FieldProjection(path = "authors.lastName") // <2>
+        List<String> authorLastNames
+) {
+}
+----
+<1> Projects to field `title` (path derived from the name of the constructor parameter)
+<2> Projects to field `authors.lastName` (explicit path)
+
+See link:{hsearch-doc-url-prefix}#mapping-projection[this section of the reference documentation]
+for more information.
+
+[[search-highlighting]]
+=== Highlighting
+
+Hibernate Search now offers a new feature in the Search DSL: highlighting.
+
+Highlighting is a projection that returns fragments from full-text fields of matched documents that caused a query match.
+Specific terms that caused the match are "highlighted" with a pair of opening and closing tags.
+It can help a user to quickly identify the information they were searching for on a results page.
+
+For example, one can enable highlighting on a full-text field like this:
+
+[source, JAVA, indent=0]
+----
+@Entity
+@Indexed
+public class Book {
+    @Id
+    private Integer id;
+
+    @FullTextField(analyzer = "english")
+    private String title;
+
+    @FullTextField(analyzer = "english", projectable = Projectable.YES)
+    private String description;
+}
+----
+
+And then retrieve highlights like this:
+
+[source, JAVA, indent=0]
+----
+SearchSession searchSession = /* ... */
+
+List<List<String>> result = searchSession.search( Book.class )
+        .select( f -> f.highlight( "description" ) )
+        .where( f -> f.match().field( "description" ).matching( "robot" ) )
+        .fetchHits( 20 );
+----
+
+Which can for example return the following highlights:
+
+[source, indent=0]
+----
+# Hit #0
+["A <em>robot</em> becomes self-aware."]
+# Hit #1
+["A <em>robot</em> helps investigate a murder on an extrasolar colony.", "On this planet, <em>robots</em> are used extensively."]
+# etc.
+----
+
+This is a simple example, but highligting can be configured extensively.
+See link:{hsearch-doc-url-prefix}#search-dsl-highlighting[this section of the reference documentation]
+for more information.
+
+[[indexing-plan-filter]]
+=== Indexing plan filters
+
+With link:{hsearch-jira-url-prefix}/HSEARCH-168[HSEARCH-168]/link:{hsearch-jira-url-prefix}/HSEARCH-1383[HSEARCH-1383],
+Hibernate Search now allow enabling/disabling indexing in indexing plans (which include in particular automatic indexing),
+both fully (for all types) or on a per-type basis.
+
+For example, the following will disable automatic indexing
```

Comment:
I am now triggered by `automatic indexing` 😆

Reply:
> I am now triggered

I see what you did here